base/linalg/triangular.jl

```diff
@@ -149,65 +149,77 @@ istril(A::UnitLowerTriangular) = true
 istriu(A::UpperTriangular) = true
 istriu(A::UnitUpperTriangular) = true
 
-function tril(A::UpperTriangular,k::Integer=0)
+function tril!(A::UpperTriangular,k::Integer=0)
     n = size(A,1)
     if abs(k) > n
         throw(ArgumentError("requested diagonal, $k, out of bounds in matrix of size ($n,$n)"))
     elseif k < 0
-        return UpperTriangular(zeros(A.data))
+        fill!(A.data,0)
+        return A
     elseif k == 0
         return UpperTriangular(diagm(diag(A)))
     else
-        return UpperTriangular(triu(tril(A.data,k)))
+        return UpperTriangular(triu!(tril!(A.data,k)))
```

Comment:
Is the `triu!` actually necessary? Doesn't `UpperTriangular` provide a view that already ignores the lower triangle of data?